pkg/pkgmgr/rpm.go

```diff
@@ -401,22 +401,25 @@ func (rm *rpmManager) installUpdates(ctx context.Context, updates unversioned.Up
 	var installCmd string
 	switch {
 	case rm.rpmTools["dnf"] != "":
-		if rm.checkForUpgrades(ctx, rm.rpmTools["dnf"]) != nil {
-			return nil, nil, errors.New("no upgradable packages")
+		checkUpdateTemplate := `sh -c "%[1]s check-update; if [ $? -ne 0 ]; then echo >> /updates.txt; fi"`
+		if !rm.checkForUpgrades(ctx, rm.rpmTools["dnf"], checkUpdateTemplate) {
+			return nil, nil, fmt.Errorf("no patchable packages found")
 		}
 
 		const dnfInstallTemplate = `sh -c '%[1]s upgrade %[2]s -y && %[1]s clean all'`
 		installCmd = fmt.Sprintf(dnfInstallTemplate, rm.rpmTools["dnf"], pkgs)
 	case rm.rpmTools["yum"] != "":
-		if rm.checkForUpgrades(ctx, rm.rpmTools["yum"]) != nil {
-			return nil, nil, errors.New("no upgradable packages")
+		checkUpdateTemplate := `sh -c 'if [ "$(%[1]s -q check-update | wc -l)" -ne 0 ]; then echo >> /updates.txt; fi'`
+		if !rm.checkForUpgrades(ctx, rm.rpmTools["yum"], checkUpdateTemplate) {
+			return nil, nil, fmt.Errorf("no patchable packages found")
 		}
 
 		const yumInstallTemplate = `sh -c '%[1]s upgrade %[2]s -y && %[1]s clean all'`
 		installCmd = fmt.Sprintf(yumInstallTemplate, rm.rpmTools["yum"], pkgs)
 	case rm.rpmTools["microdnf"] != "":
-		if rm.checkForUpgrades(ctx, rm.rpmTools["microdnf"]) != nil {
-			return nil, nil, errors.New("no upgradable packages")
+		checkUpdateTemplate := `sh -c "%[1]s install dnf -y; dnf check-update -y; if [ $? -ne 0 ]; then echo >> /updates.txt; fi;"`
```

Comment:
can we use the sh function added in #618

Reply:
the command does not work unless we specify "sh -c" in the argument in addition to llb.Shlex. I am not sure why, but it is the same format followed in the installCmd.

Reply:
Ok let's follow up after if we can get this working